### PR TITLE
Added the Prompt component

### PR DESCRIPTION
**Pull request by sayantan1413**
Signed-off-by: sayantan1413 <sayantanbose2001@gmail.com>

**Description**

Added the Prompt component

This PR fixes #

**Notes for Reviewers**

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [ ] Yes, I signed my commits.

### CODE REVIEW

File: ui/components/Prompt/Prompt.js [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+const Prompt = () => {
+  const classes = useStyles();
+  const [showModal, setShowModal] = useState(false);
+  const [title, setTitle] = useState("");
+  const [subtitle, setSubtitle] = useState("");
+  const [options, setOptions] = useState([]);
+  let promiseInfo = {};
+
+  const show = async (passed) => {
```
Comment (sudo-NithishKarthik) [inline ui/components/Prompt/Prompt.js:21]: This function need not be async

File: ui/components/Prompt/Prompt.js [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+  Button,
+  Dialog,
+  DialogActions,
+  DialogContentText,
+  DialogContent,
+  DialogTitle,
+} from "@mui/material";
+import { useStyles } from "./Prompt.styles.js";
+
+const Prompt = () => {
```
Comment (sudo-NithishKarthik) [inline ui/components/Prompt/Prompt.js:13]: We should be getting `title` `subtitle` and `options` from the props

File: ui/components/Prompt/Prompt.js [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+          </DialogTitle>
+        )}
+        {subtitle !== "" && (
+          <DialogContent>
+            <DialogContentText id="alert-dialog-description" className={classes.subtitle}>
+              <Typography variant="body1">{subtitle}</Typography>
+            </DialogContentText>
+          </DialogContent>
+        )}
+        <DialogActions className={classes.actions}>
```
Comment (sudo-NithishKarthik) [inline ui/components/Prompt/Prompt.js:57]: The number of buttons rendered should be dynamic, i.e it should render a button for every option provided.

**Review by sudo-NithishKarthik — CHANGES_REQUESTED** [review]
Good work @sayantan1413 . There are some modifications to be done. I think you will have to rethink how we can dynamically display the number of options. That would be good challenge also, hope you enjoy 😄

File: ui/components/Prompt/Prompt.js [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+} from "@mui/material";
+import { useStyles } from "./Prompt.styles.js";
+
+const Prompt = () => {
+  const classes = useStyles();
+  const [showModal, setShowModal] = useState(false);
+  const [title, setTitle] = useState("");
+  const [subtitle, setSubtitle] = useState("");
+  const [options, setOptions] = useState([]);
+  let promiseInfo = {};
```
Comment (bariqhibat) [inline ui/components/Prompt/Prompt.js:19]: Won't this be good to be used as a state instead?

File: ui/components/Prompt/Prompt.js [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+            </DialogContentText>
+          </DialogContent>
+        )}
+        <DialogActions className={classes.actions}>
+          <Button
+            onClick={() => {
+              hide();
+              resolve(options[1]);
+            }}
+            key={options[1]}
```
Comment (bariqhibat) [inline ui/components/Prompt/Prompt.js:63]: Shouldn't we check the length of this array `options` first? If it's empty then this will throw an error

**Review by bariqhibat — CHANGES_REQUESTED** [review]
Some comments

**Comment by leecalcote** [comment]
@sudo-NithishKarthik, @sayantan1413 has stepped away from this effort. 

Will you carry this PR to conclusion (merge, fix, close)?

**Comment by sudo-NithishKarthik** [comment]
Yes. Will do 👍

**Comment by sudo-NithishKarthik** [comment]
Opening a new PR for this.

**Pull request closed without merging**